_pages/dev/tag-management-system.md

```diff
@@ -0,0 +1,220 @@
+    window.myDataLayer.push(event);
+  }
+}
+```
+
+After you have written your collector, you configure it, as shown in the following example:
+
+```typescript
+BaseTmsModule.forRoot({
+  tagManager: {
+    gtm: {
+      collector: MyCollectorService,
+      events: [NavigationEvent, CartAddEntrySuccessEvent],
+    },
+    ...
+  },
+}),
+```
+
+### Events Payload
+
+Depending on which TMS client you are working with, the requirements for an event's payload can vary a lot. The default event payloads in Spartacus are intended to match the requirements for the most typical use cases. However, if the defaults do not meet your needs, you can re-map or adjust the payloads in a few different ways, as described in the following sections.
+
+#### Creating a Custom Mapper
+
+To create a custom mapper for a specific TMS solution, you can create a custom collector (as shown in the [Data Layer](#data-layer) section, above), and then implement the `map` method, as shown in the following example:
+
+```typescript
+import { Injectable } from "@angular/core";
+import { CxEvent } from "@spartacus/core";
+import {
+  TmsCollector,
+  TmsCollectorConfig,
+  WindowObject,
+} from "@spartacus/tracking/tms/core";
+
+@Injectable({ providedIn: "root" })
+export class MyCollectorService implements TmsCollector {
+  ...
+
+  map?<T extends CxEvent>(event: T): T | object {
+    // TODO: custom mapping logic goes here...
+  }
+}
+```
+
+In the `map` method, you can use the `instanceof` check to indicate which events should be re-mapped. If you find this logic becoming too complex, you can inject other services to your `MyCollectorService`, as needed.
+
+#### Creating Custom Events
+
+You can create a custom event in Spartacus and re-map the data to fit your data structure requirements simply by registering a new event source, as described in [Registering Event Sources]({{ site.baseurl }}/event-service/#registering-event-sources). The following is an example:
+
+```typescript
+const eventSource = this.eventService.get(NavigationEvent).pipe(
+  map((navigationEvent) =>
+    createFrom(CustomNavigationEvent, {
+      // the next lines are example logic, not necessarily the actual page name and type
+      pageName: navigationEvent.context.id,
+      pageType: navigationEvent.context.type,
+    })
+  )
+);
+
+eventService.register(CustomEvent, eventSource);
+```
+
+In the example above, the Spartacus `NavigationEvent` is re-mapped to a `CustomNavigationEvent`. If you need to pull additional data, see [Pulling Additional Data From Facades]({{ site.baseurl }}/event-service/#pulling-additional-data-from-facades) for more information.
+
+**Note:** To bootstrap the logic, you need to inject your `CustomNavigationEventBuilder` somewhere, such as in a dummy module that is being imported, for example, to the `AppModule`.
+
+If you use this approach, you need to pass your `CustomNavigationEvent` to the `tms.adobeLaunch.events` config array, instead of the default `NavigationEvent`.
```

Comment:
Based on our offline discussion, we noticed that this should be changed to `tms.aep.events`

Reply:
Sentence updated as follows:

> If you use this approach, you need to pass your `CustomNavigationEvent` to the relevant config array (such as `tagManager.gtm.events` or `tagManager.aep.events`), instead of the default `NavigationEvent`.